pkg/asset/manifests/gcp/cluster.go

```diff
@@ -0,0 +1,138 @@
+		description       = "Created By OpenShift Installer"
+	)
+
+	networkName := fmt.Sprintf("%s-network", clusterID.InfraID)
+	if installConfig.Config.GCP.Network != "" {
+		networkName = installConfig.Config.GCP.Network
+	}
+
+	masterSubnet := generateSubnetName(clusterID.InfraID, "master")
+	if subnet := installConfig.Config.GCP.ControlPlaneSubnet; subnet != "" {
+		masterSubnet = subnet
+	}
+
+	master := capg.SubnetSpec{
+		Name:           masterSubnet,
+		CidrBlock:      "",
+		Description:    ptr.To(description),
+		Region:         installConfig.Config.GCP.Region,
+		EnableFlowLogs: ptr.To(flowLogsEnabled),
+	}
+
+	workerSubnet := generateSubnetName(clusterID.InfraID, "worker")
+	if subnet := installConfig.Config.GCP.ComputeSubnet; subnet != "" {
+		workerSubnet = subnet
+	}
+
+	worker := capg.SubnetSpec{
+		Name:           workerSubnet,
+		CidrBlock:      "",
+		Description:    ptr.To(description),
+		Region:         installConfig.Config.GCP.Region,
+		EnableFlowLogs: ptr.To(flowLogsEnabled),
+	}
+
+	// Add the CIDR information.
+	machineV4CIDRs := []string{}
+	for _, network := range installConfig.Config.Networking.MachineNetwork {
+		if network.CIDR.IPNet.IP.To4() != nil {
+			machineV4CIDRs = append(machineV4CIDRs, network.CIDR.IPNet.String())
+		}
+	}
+
+	if len(machineV4CIDRs) == 0 {
+		return nil, fmt.Errorf("failed to parse machine CIDRs")
+	}
+
+	_, ipv4Net, err := net.ParseCIDR(machineV4CIDRs[0])
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse machine network CIDR: %w", err)
+	}
+
+	if installConfig.Config.GCP.ControlPlaneSubnet == "" {
+		masterCIDR, err := cidr.Subnet(ipv4Net, 1, 0)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create the master subnet %w", err)
+		}
+		master.CidrBlock = masterCIDR.String()
+	}
+
+	if installConfig.Config.GCP.ComputeSubnet == "" {
+		workerCIDR, err := cidr.Subnet(ipv4Net, 1, 1)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create the worker subnet %w", err)
+		}
+		worker.CidrBlock = workerCIDR.String()
+	}
+
+	subnets := []capg.SubnetSpec{master, worker}
+
+	labels := map[string]string{}
+	labels[fmt.Sprintf("kubernetes-io-cluster-%s", clusterID.InfraID)] = "owned"
+	labels[fmt.Sprintf("capg-cluster-%s", clusterID.InfraID)] = "owned"
+	for _, label := range installConfig.Config.GCP.UserLabels {
+		labels[label.Key] = label.Value
+	}
+
+	gcpCluster := &capg.GCPCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      clusterID.InfraID,
+			Namespace: capiutils.Namespace,
+		},
+		Spec: capg.GCPClusterSpec{
+			Project: installConfig.Config.GCP.ProjectID,
+			Region:  installConfig.Config.GCP.Region,
+			Network: capg.NetworkSpec{
+				// TODO: Need a network project for installs where the network resources will exist in another
+				// project such as shared vpc installs
+				Name:                  ptr.To(networkName),
+				AutoCreateSubnetworks: ptr.To(autoCreateSubnets),
+				Subnets:               subnets,
+			},
+			AdditionalLabels: labels,
```

Comment:
Just wondering if `FailureDomains` needs to be set based on the value of mpool.Zones or if  `FailureDomains` will always be empty, i.e. default to all of the zones in the selected region.

Reply:
> Just wondering if FailureDomains needs to be set based on the value of mpool.Zones

It does, users can use the mpool.Zones to choose specific zones.